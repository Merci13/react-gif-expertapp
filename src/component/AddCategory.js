

import React, { useState } from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ( {setCategories} ) => {

let [inputValue, setImputValue] = useState('');


   const handleInputChange = ( e ) =>{
        setImputValue(e.target.value);
  
   } ;

   const handleSummit = ( e ) =>{
        e.preventDefault();
        if( inputValue.trim().length > 2){
              setCategories(categories => [inputValue, ...categories]);

        console.log('Summit Hecho!');
        setImputValue ("");
        }

      
   };

    return (
        <>
 
            <form onSubmit = {handleSummit}>
                  <input
            type= 'text'
            value = {inputValue}
            onChange ={ handleInputChange }
         />
            </form>
       
        </>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired

};
