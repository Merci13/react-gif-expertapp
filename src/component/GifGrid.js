
import React from 'react'
import { useFecthGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category = "" }) => {
   // const apyKey = 'wGlqhmQl8UGQDP7A0JnHkadJ1MYFb10X';





  
   
   const {data:images, loading} = useFecthGifs( category);
  



    return ( 
        <> 
        <h3 className="animate__fadeIn">{category}</h3>

        {loading && <p className= "animate__pulse"> Cargando</p>}

        <div className ="card-grid">
           

                {
                images.map((image) => (
                        // <li key = {image.id}> {image.title}</li>
                        <GifGridItem 
                        key ={image.id}
                        {...image}
                        />
                ))
                }

        </div>
        </>
   
    )
}
