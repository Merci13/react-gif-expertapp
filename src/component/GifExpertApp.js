
import React from 'react';


//import PropTypes from 'prop-types';
import { useState } from 'react/cjs/react.development';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = () =>{

    //const categories = ['One punch', 'Samura X', 'Dragon Ball'];

    const [categories, setCategories] = useState(['One punch']);


    // const handleAdd = () =>{

    //    // setCategories( [...categories, 'HunterxHunter']);
    //   //  setCategories(categories => [...categories, 'HunterxHunter']);
    
    // };

    return (
        <>
        <h2>GifExpertApp</h2>
        <hr/>
        <AddCategory setCategories = { setCategories }/>

        <hr/>

       

        <ol>
            { 
                categories.map( (category, index) => 
         ( 
         <GifGrid 
         key = {category + index}
         category = {category}/>))
            }

        </ol>

        </>
    );
};



export default GifExpertApp;