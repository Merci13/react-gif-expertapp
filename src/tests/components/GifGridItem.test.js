
import React from 'react'

import { shallow } from 'enzyme';

import { GifGridItem } from "../../component/GifGridItem";



describe('Test suit for GifGridItem', () => {
    const data = {
        id: "1",
        title: "Test",
        url: 'https://media0.giphy.com/media/frSfC5NcmyF7q/giphy.gif?cid=06e8b66c72sm7qbmymoflqopm25mpd9kjc5or58kdq34eljb&rid=giphy.gif&ct=g'
    };

    const wrapper = shallow(< GifGridItem id={data.id} title={data.title} url={data.url} />);


    test('El GifGridItem tiene que verse', () => {



        expect(wrapper).toMatchSnapshot();


    });

    test('Debe de tener un parrafo con el titulo', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(data.title);

    });

    test('Debe de tener la image igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(data.url);
        expect(img.prop('alt')).toBe(data.title);



    });


    test('Debe de tener animate__fadeIn', () => {

        const div = wrapper.find('div');
        expect(div.prop('className').includes("animate__fadeIn")).toBe(true);

    })




});