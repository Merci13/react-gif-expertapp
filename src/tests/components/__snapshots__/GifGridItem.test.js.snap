// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suit for GifGridItem El GifGridItem tiene que verse 1`] = `
<div
  className="animate__fadeIn"
>
  <img
    alt="Test"
    src="https://media0.giphy.com/media/frSfC5NcmyF7q/giphy.gif?cid=06e8b66c72sm7qbmymoflqopm25mpd9kjc5or58kdq34eljb&rid=giphy.gif&ct=g"
  />
  <p>
     
    Test
  </p>
</div>
`;
